### PR TITLE
firefox support

### LINKED ISSUE #3: Firefox support
Extend the extension to Mozilla Firefox.

### PR DESCRIPTION
**Pull request by sahil-shubham**
This is the new pull request. This fixes #3

### CODE REVIEW

**Comment by sahil-shubham** [comment]
@thealphadollar I would let you know when I fix the issue with the security question. I should be done by tomorrow.

**Comment by sahil-shubham** [comment]
@thealphadollar I have fixed the issue with the fetching of the security question, I didn't need to do much other than adding a new permission :sweat_smile:. 

The issue that remains  is that of with the **innerHTML** which makes the "Your credentials have been saved" message pop up before all the information is put in.

**Review by thealphadollar — COMMENTED** [review]
This looks cool. Can you help me understand how we will import this into firefox? And also let me know if you can add the same in the README.md as instructions for Firefox platform.

PS. It would be an amazing experience trying to create [a Firefox](https://extensionworkshop.com/documentation/publish/package-your-extension/) and a Chrome extension package.

**Comment by sahil-shubham** [comment]
I have added the .xpi file, you can add that in the addons.

>  if you can add the same in the README.md as instructions for the Firefox platform.

Sure, I would do that.

**Comment by sahil-shubham** [comment]
@thealphadollar Please review and let me know of any changes, Thank you.

**Review by thealphadollar — APPROVED** [review]
Voohoo! This is awesome! Thanks a lot, Sahil, for adding support for firefox. It was such a highly needed feature.

BTW, I was wondering what are the steps you used to create the `xpi`, do you think adding it to `CONTRIBUTING.md` with the steps would help others create it? It'll make it quick for us to release a new firefox version with new features and fixes, no?

I'm sure you'd have taken this into account, just to be sure, you've checked that this is still working with Chrome after the changes right? :fearful:

**Comment by sahil-shubham** [comment]
> I'm sure you'd have taken this into account and checked that this is still working with Chrome after the changes, right?

Yes, I have checked it still works with chrome.

> do you think adding it to CONTRIBUTING.md with the steps would help others create it?

I think it would be nice addition to the file. I would add the steps.

**Comment by sahil-shubham** [comment]
@thealphadollar I have added the steps to make a .xpi file. Give it a look whenever you are able to find time, Thank you

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -37,6 +37,21 @@ It is highly recommended to follow the below guidelines while writing commits an
...
+
+2. Enter this command to compress the files into a .zip file, while excluding the unnecessary files -
+   
+   `zip -r  erp-auto-login_ff.zip * -x "*.git*" "instruction_images/*"`
+
+3. Now enter the following command to rename the .zip file to a .xpi file -
+
+    `mv erp-auto-login_ff.zip erp-auto-login_ff.xpi`
+
+4.  You have successfully made your extension into a xpi file :)
```
Comment (thealphadollar) [inline CONTRIBUTING.md:54]:
```suggestion
4.  You have successfully packed the extension into a xpi file :)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -37,6 +37,21 @@ It is highly recommended to follow the below guidelines while writing commits an
 - https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/ [*[archive]*](http://archive.is/BbIbh)
 - https://code.likeagirl.io/useful-tips-for-writing-better-git-commit-messages-808770609503 [*[archive]*](http://archive.is/W1h2O)
 
+## How to make a .xpi file
```
Comment (thealphadollar) [inline CONTRIBUTING.md:40]:
```suggestion
## How to make Firefox addon package (.xpi) file
```

File: README.md [hunk 3]
```diff
@@ -2,7 +2,20 @@ IITKGP ERP AUTO LOGIN
 =====================
 A chrome extension that has to be set up just one time for automatic login to your IITKGP ERP account every time. Just [install the chrome extension](https://chrome.google.com/webstore/detail/iitkgp-erp-auto-login/gdickphilmdekcbmpjmbnbikchaecbdk), and click on the extension logo (IIT Kharagpur's logo) to set up your login credentials.
 
-<img src="demo.png">
+<img src="instruction_images/demo.png">
+
+## How to add this extension to Firefox
+
+This extension can aslo be added to Firefox. The steps to follow for the same are:
```
Comment (thealphadollar) [inline README.md:9]:
```suggestion
This extension can also be added to Firefox. The steps to follow for the same are:
```

**Review by thealphadollar — APPROVED** [review]
This is done! Thanks @sahil-shubham for closing a crucial fix. We've had so many students asking for Firefox version.

I've suggested three minor changes (just nitpicking). Once you have committed them, I'll merge the PR.

**Comment by sahil-shubham** [comment]
@thealphadollar Thank you for going through my changes, I would try to catch spelling mistakes on my own from next time. I learned a lot of new things because of this issue.

**Comment by thealphadollar** [comment]
Please leave some work for us reviewers too 😖 Your PRs are already so good, and phir ye bhi nhi krne ko milega 😭

**Pull request merged**